docs/containers/testing-with-docker.md

```diff
@@ -0,0 +1,51 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Testing with docker](#testing-with-docker)
+  - [Using the included docker-compose support](#using-the-included-docker-compose-support)
+    - [Usage](#usage)
+    - [Configuration](#configuration)
+    - [Notes](#notes)
+    - [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Testing with docker
+
+## Using the included docker-compose support
+
+### Usage
+
+```bash
+docker build -t lyft/cartography
+docker-compose up -d
+docker-compose run cartography ...
+```
+
+### Configuration
+
+Configuration is possible via the `.compose` directory, which is
+git ignored. Environment for cartography is in
+`.compose/cartography/cartography.env`. neo4j config, logs, etc is
+located at `.compose/neo4j/...`
+
+### Notes
+
+* On initial start of the compose stack, it's necessary to
+change the neo4j user's password through the neo4j UI.
+* Neither the docker image, nor the docker-compose file define an
+entrypoint, so it's necessary to pass in the command being run. This
+also makes it possible to run a custom sync script, rather than only
+cartography.
+
+### Example
+
+```bash
+# See the cartography github configuration intel module docs
+export GITHUB_KEY=BASE64ENCODEDKEY
```

Comment:
I thought about this a bit more and I'm a bit concerned about asking a user to put their creds in the terminal which would get recorded in bash history.

AWS official docs [don't bother](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials_environment.html) to deal with this (I suppose they assume the dev knows this), but since this is a security tool I feel like we should do better.

How do you feel about adding a disclaimer instructing the user to turn off history logging temporarily when setting creds with `set +o history` and then turning it on back again with `set -o history` ([ref](https://attack.mitre.org/techniques/T1552/003/))?

Reply:
I added disabling/renabling command history to the example.

I think it's good we're being good security citizens here, but IAM creds and github access keys normally live in files in the user's home directory, but no need to have them live in other spots on the filesystem :D